Lib/asyncio/queues.py

```diff
@@ -125,20 +151,28 @@ async def put(self, item):
                     self._putters.remove(putter)
                 except ValueError:
                     # The putter could be removed from self._putters by a
-                    # previous get_nowait call.
+                    # previous get_nowait call,
+                    # or a shutdown call.
                     pass
                 if not self.full() and not putter.cancelled():
                     # We were woken up by get_nowait(), but can't take
                     # the call.  Wake up the next in line.
                     self._wakeup_next(self._putters)
                 raise
+            if not self._is_alive():
+                raise QueueShutDown
         return self.put_nowait(item)
 
     def put_nowait(self, item):
         """Put an item into the queue without blocking.
 
         If no free slot is immediately available, raise QueueFull.
+
+        Raise a QueueShutDown if _shutdown_state is not set to
+        `ALIVE`.
```

Comment:
```suggestion
        Raises QueueShutDown if the queue has been shut down.
```

Doc-strings are for humans